actor/builtin/miner/miner.go

```diff
@@ -714,3 +723,75 @@ func (ma *Actor) GetProvingPeriodStart(ctx exec.VMContext) (*types.BlockHeight,
 
 	return state.ProvingPeriodStart, 0, nil
 }
+
+func (ma *Actor) currentProvingPeriodPoStChallengeSeed(ctx exec.VMContext) (proofs.PoStChallengeSeed, error) {
+	chunk, err := ctx.ReadStorage()
+	if err != nil {
+		return proofs.PoStChallengeSeed{}, err
+	}
+
+	var state State
+	if err := actor.UnmarshalStorage(chunk, &state); err != nil {
+		return proofs.PoStChallengeSeed{}, err
+	}
+
+	bytes, err := ctx.SampleChainRandomness(state.ProvingPeriodStart)
+	if err != nil {
+		return proofs.PoStChallengeSeed{}, err
+	}
+
+	seed := proofs.PoStChallengeSeed{}
+	copy(seed[:], bytes)
+
+	return seed, nil
+}
+
+// SampleChainRandomness produces a slice of random bytes sampled from a TipSet
+// in the blockchain at a given height (minus lookback). This is useful for
+// things like PoSt challenge seed generation.
+//
+// If no TipSet exists with the provided height (sampleHeight), we instead
```

Comment:
This comment tells lies and I will fix it. If sample height minus lookback is negative, I sample from genesis block. If sample height minus lookback is not negative and no tip set with matching height exists in channel, I produce an error.